lib/ex_aws/sqs/client.ex

```diff
@@ -0,0 +1,122 @@
+defmodule ExAws.SQS.Client do
+  use Behaviour
+
+  @moduledoc """
+  The purpose of this module is to surface the ExAws.SQS API tied to a single
+  configuration chose, such that it does not need passed in with every request
+
+  Usage:
+  ```elixir
+  defmodule MyApp.SQS do
+    use ExAws.SQS.Client, otp_app: :my_otp_app
+  end
+  ```
+
+  In your config
+  ```elixir
+  config :my_otp_app, ExAws,
+    sqs: [], #SQS config goes here
+  ```
+  """
+
+  @type sqs_permission :: 
+    :all | 
+    :receive_message | 
+    :delete_message |
+    :change_message_visibility | 
+    :get_queue_attributes 
+  @type sqs_acl :: [{binary, sqs_permission}]
+  @type sqs_message_attribute_name ::
+    :all |
+    :sender_id | 
+    :sent_timestamp |
+    :approximate_receive_count |
+    :approximate_first_receive_timestamp |
+    :wait_time_seconds |
+    :receive_message_wait_time_seconds 
+  @type sqs_queue_attribute_name ::
+    :all | 
+    :approximate_number_of_messages |
+    :approximate_number_of_messages_not_visible | 
+    :visibility_timeout |
+    :created_timestamp | 
+    :last_modified_timestamp | 
+    :policy  
+  @type sqs_message :: [
+    {:message_body, binary},
+    {:delay_seconds, 0..900 | none},
+    {:attributes, Keyword.t }
+  ]
```

Comment:
this doesn't compile.
```elixir
@type sqs_message :: [
  {:message_body, binary},
  {:delay_seconds, 0..900} |
  {:attributes, Keyword.t } |
  none
]
```
After reading the documentation closer, I'm like this better:
```elixir
  @type sqs_message_attribute :: %{
    :data_type => :string | :number | :binary | binary,
    :value => binary | number
  }
  @type sqs_message :: %{:message_body => binary} |
  %{:message_body => binary, :delay_seconds => 0..900} | 
  %{:message_body => binary, :attributes => [sqs_message_attribute, ...] | sqs_message_attribute } |
  %{:message_body => binary, :delay_seconds => 0..900, :attributes => [sqs_message_attribute, ...] | sqs_message_attribute }
```
Although I'm not totally sold on the attribute map inside the message map, but it does work.

Thoughts?